docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -1,326 +1,752 @@
+Stream<Person> persons = personRepository.streamAll();
 List<String> namesButEmmanuels = persons
     .map(p -> p.name.toLowerCase() )
     .filter( n -> ! "emmanuel".equals(n) )
     .collect(Collectors.toList());
 ----
 
-NOTE: The `stream` methods require a transaction to work.
+== Hibernate ORM with Panache queries
+
+Hibernate Query Language (HQL) is a query language similar to Structured Query Language (SQL). However, HQL works with persistent objects and properties instead of tables and columns. For more information about HQL, see the "HQL and JPQL" section of the xref:https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html[_Hibernate ORM User Guide_].
+
+=== PanacheQuery entity
+
+The `PanacheQuery` entity  abstracts the use of paging, getting the number of results, and operating on `List` or `Stream` methods. You can use the `PanacheQuery` entity  for paging and range-based queries.
 
 
-== Paging
+Only use `List` and `Stream` methods if your table contains small data sets. For large data
+sets, use the `Find` method equivalents, which return a `PanacheQuery` on which you can perform paging.
 
-You should only use `list` and `stream` methods if your table contains small enough data sets. For larger data
-sets you can use the `find` method equivalents, which return a `PanacheQuery` on which you can do paging:
+[WARNING]
+====
+You cannot mix ranges and pages. If your query uses a range, any methods that depend on having a current page throw an `UnsupportedOperationException` exception. To switch back to paging, use `page(Page)` or `page(int, int)`.
+====
 
+.Additional resources
+* For a complete list of  `PanacheQuery` methods, see the https://javadoc.io/doc/io.quarkus/quarkus-hibernate-orm-panache/latest/io/quarkus/hibernate/orm/panache/PanacheQuery.html[_PanacheQuery Class_] page.
+
+==== Examples of queries that use paging
+These examples show how you can use paging in your Panache queries. The active record pattern examples use a `Person` entity created from the `PanacheEntity` class. The repository pattern examples use a regular JPA `Person` entity and the `PersonRepository` created from the
+`PanacheRepository` class.
+
+
+
+* Create a query for all living persons:
+** Active record pattern:
++
 [source,java]
 ----
-// create a query for all living persons
 PanacheQuery<Person> livingPersons = Person.find("status", Status.Alive);
+----
+** Repository pattern:
++
+[source,java]
+----
+PanacheQuery<PersonRepository> livingPersons = Person.find("status", Status.Alive);
+----
+
 
-// make it use pages of 25 entries at a time
+* Make pages have 25 entries per page:
++
+[source,java]
+----
 livingPersons.page(Page.ofSize(25));
+----
+
 
-// get the first page
+* Retrieve the first page:
+** Active record pattern:
++
+[source,java]
+----
 List<Person> firstPage = livingPersons.list();
+----
+** Repository pattern:
++
+[source,java]
+----
+List<Person.Repository> firstPage = livingPersons.list();
```

Comment:
```suggestion
List<Person> firstPage = livingPersons.list();
```

Here the two styles are equivalent: once you have a `PanacheQuery` its usage doesn't differ between the two styles: they only differ in how you obtain the `PanacheQuery` in the first place.